src/rgw/rgw_admin.cc

```diff
@@ -8320,6 +8347,26 @@ int main(int argc, const char **argv)
     }
   }
 
+  if (opt_cmd == OPT::BUCKET_CHECK_OLH) {
+    if (!dynamic_cast<rgw::sal::RadosStore*>(driver)) {
+      cerr <<
+	      "WARNING: this command is only relevant when the cluster has a RADOS backing store." <<
+	      std::endl;
+      return 0;
+    }
+    RGWBucketAdminOp::check_index_olh(driver, bucket_op, stream_flusher, dpp());
```

Comment:
since you checked the cast for `RadosStore` here, you might as well pass the casted variable into `check_index_olh()` so it doesn't have to cast again